### ISSUE TITLE
Allow overriding/adding any requirejs parameter

### ISSUE THREAD

**Issue by rockResolve** [welldone-software/gulp-durandal#1]
I want to be able to set the pragmas property on the rjsConfig object. But it seems a bit pointless to keep on adding gulp_durandal properties for every requireJs property anyone requests. (I messed around with quite a few properties before settling on pragmas).

So how about a global requirejs object which can override or add any requirejs property.

I would use like:
    durandal({
            ...
        requirejs: {
            pragmas: {
                prodBuild: true
            },
        }
    })

**Comment by mryellow** [comment]
Would be nice if options like `generateSourceMaps` and customisations to `uglify2` config were passed through from the durandal constructor. Could hit all the stuff documented here then leave the function for anything out of the ordinary:

https://github.com/jrburke/r.js/blob/master/build/example.build.js

**Comment by danikenan** [comment]
Please look at options.rjsConfigAdapter.

**Comment by mryellow** [comment]
Yeah it can do all that, but it would be more intuitive for these options to be passed through by default.

**Comment by danikenan** [comment]
Sorry, but you're missing the whole point of the plugin.

1. The idea is to _hide_ the complexity of rjs and provide something
minimalistic, that fits 90% of the cases using 0-2 simple config options.
2. The options we offer are in our own vocabulary, not that of
rjs/requirejs.
3. You propose to mix vocabularies. Mix rjs and none rjs options. This
seems a bad idea, certainly not a "more intuitive" one.
4. We want to promote the simplified usage, not the complex rjs usage.
5. The plugin follows our motto: keep simple things simple, ensure complex
things are possible.

In short, we prefer a simple usage and syntax for the 90% who are not
familiar with complex rjs parameters, and make things intuitive for them.
We enabled complete freedom with the config adapter.
What more???

**Comment by mryellow** [comment]
The keyword here is "intuitive", Grunt and other pieces that work with RequireJS often configure it by passing on the config objects on directly, this sets up a paradigm which means when arriving at gulp-durandal people can be a little misdirected when they see the options, as testimonied by this thread and others, multiple confused people.

Think it's just a tiny documentation issue.

When a new user sees the options available, it sets up a familiar picture "these all look like RequireJS options, everything else just passes them through, I'll just add some extra options like I've done before with other libraries".

```
function(rjsConfig){
  rjsConfig.pragmas = {
        fooExclude: true
  };
  return rjsConfig;
}
```

Fixing this example up to be a real example would be clearer. 

```
function(cfg) {
        cfg.preserveLicenseComments = false;
        cfg.generateSourceMaps = false;
        cfg.uglify2 = {
            output: {
                beautify: false
            },
            compress: {
                sequences: false,
                global_defs: {
                    DEBUG: false
                },
                drop_console: true
            },
            warnings: true,
            mangle: false
        };

        return cfg;
    }
})
```

**Comment by danikenan** [comment]
We are not a requirejs plugin, but a Durandal one.

**Comment by mryellow** [comment]
Well guess the documentation doesn't need clearing up then, no issue and no one has ever been confused by it.

**Comment by danikenan** [comment]
We have ~ 1000 d/l per month, for more than a year.
No one else complained...

**Comment by mryellow** [comment]
Righto... Writing clearer docs that better explain your systems is never a good idea.

Back to counting downloads.

**Comment by danikenan** [comment]
I fact, we added quite a few features requested by others and
even incorporated several pull requests of code and changes.

We simply do not agree with you on the features you wanted at beginning, as
we find them promoting bad practices (mixing vocabularies as well as not
hiding the rjs complexity). Also, we do not agree  the intuitiveness of you
request, we find it contra-intuitive.

This is our view, like it or not.

We feel that your expectations are based on  your familiarity with other
requirejs and durandal plugins in grunt, which we found wrong to begin with
(that is why we created our own).

Never the less, you are more than welcome to propose an improvement to the
docs that will help people like you (with the same state of mind and
expectations) better understand how to use this plugin. If its good, rest
assure we add you doc improvements.

**Comment by mryellow** [comment]
You made the mixed vocab point well and I agree.

At which stage it became a documentation issue... but it's ok, I've had a cookie now. ;-)
